MIGRATIONS.md

```diff
@@ -0,0 +1,117 @@
+# Snuba migrations field guide
```

Comment:
Would it be worth moving this file in the migrations module ?

Reply:
I thought about this; ultimately left it here because some stuff (cli) is still outside outside that module, and it seemed easier to find here.